tests/foreman/cli/test_realm.py

```diff
@@ -25,17 +25,32 @@
 from robottelo.cleanup import realm_cleanup
 from robottelo.cli.base import CLIReturnCodeError
 from robottelo.cli.factory import CLIFactoryError
+from robottelo.cli.factory import create_object
 from robottelo.cli.factory import make_proxy
 from robottelo.cli.factory import make_realm
+from robottelo.cli.proxy import CapsuleTunnelError
+from robottelo.cli.proxy import Proxy
 from robottelo.cli.realm import Realm
+from robottelo.helpers import default_url_on_new_port
+from robottelo.helpers import get_available_capsule_port
+from robottelo.logging import logger
 
 
 @pytest.fixture(scope='module')
-def _make_proxy(request):
+def _make_proxy(request, default_sat):
     """Create a Proxy and register the cleanup function"""
-    proxy = make_proxy()
-    request.addfinalizer(lambda: capsule_cleanup(proxy['id']))
-    return proxy
+    args = {'name': gen_string(str_type='alpha')}
+    newport = get_available_capsule_port()
+    try:
+        with default_url_on_new_port(9090, newport) as url:
+            args['url'] = url
+            logger.debug('creating proxy object')
```

Comment:
at the moment this is the last line I see in the debug, and then it gets stuck at a very odd place and I never see the log message from inside `create_object`. 

The traceback makes it look like it went straight over these calls and is trying to do cleanup.

```Traceback (most recent call first):
  File "/home/mshriver/repos/satqe/robottelo/tests/foreman/cli/test_realm.py", line 51, in _make_proxy
    capsule_cleanup(proxy['id'])
  <built-in method next of module object at remote 0x7fe919160090>
  File "/home/mshriver/repos/satqe/robottelo/.robottelo/lib64/python3.8/site-packages/_pytest/fixtures.py", line 1181, in call_fixture_func
```